src/components/certificates/CertificateForm.tsx

```diff
@@ -25,6 +31,13 @@ const NewCertificateForm: React.FC = () => {
 
   const [formState, setFormState] = useState<Certificate>(initialFormState);
   const [formError, setFormError] = useState<string>('');
+  const { refetch } = useContext(CertificatesContext)!;
+
+  useEffect(() => {
+    if (certificate) {
+      setFormState(certificate);
+    }
+  }, [certificate]);
```

Comment:
This is not needed. You don't want that your form data chanes anytime certificate from prop change. That means you are expecting that this certificate from prop changes all the time, what is wrong. What comes from props should be just set as initial state and not changed anymore.
You should not even give a option that it changes and reset the form with some data. Once this component rendered and initial data is set the only change user can make